src/features/postsSlice.ts

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+export interface PostsState {
+  loaded: boolean;
+  hasError: boolean;
+  items: Post[];
+}
+
+const initialState: PostsState = {
+  loaded: false,
+  hasError: false,
+  items: [] as Post[],
+};
+
+export const loadPosts = createAsyncThunk(
+  'posts/fetch',
+  async (userId: number) => {
+    const posts = await getUserPosts(userId);
+
+    return posts;
+  },
+);
+
+export const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    resetPosts: state => {
+      state.loaded = false;
+      state.hasError = false;
+      state.items = [];
+    },
+  },
+  extraReducers: builder => {
+    builder
+      .addCase(loadPosts.pending, state => {
+        state.loaded = false;
```

Comment:
It's a good practice to also reset the `hasError` state here, similar to how it's done in the `commentsSlice` and `usersSlice`. This prevents a stale error from a previous failed request from being displayed while a new one is in progress.

```suggestion
state.loaded = false;
state.hasError = false;
```